app/models/package_manager/maven.rb

```diff
@@ -134,5 +138,34 @@ def self.project_name(name)
         artifactId: sections[1]
       }
     end
+
+    def self.get_pom(group_id, artifact_id, version, seen=[])
+      seen << [group_id, artifact_id, version]
+
+      xml = get_xml(
+        "http://repo1.maven.org/maven2/#{group_id}/#{artifact_id}/#{version}/#{artifact_id}-#{version}.pom"
+      )
+
+      next_group_id = xml.locate('distributionManagement/relocation/groupId').first&.nodes&.first || group_id
+      next_artifact_id = xml.locate('distributionManagement/relocation/artifactId').first&.nodes&.first || artifact_id
+      next_version = xml.locate('distributionManagement/relocation/version').first&.nodes&.first || version
+
+      if !seen.include?([next_group_id, next_artifact_id, next_version])
+        get_pom(next_group_id, next_artifact_id, next_version, seen)
+      else
+        xml
+      end
+    end
+
+    def self.licenses(xml)
+      xml_licenses = xml
+        .locate('licenses/license/name')
+        .flat_map(&:nodes)
+      return xml_licenses if xml_licenses.any?
+
+      LICENSE_STRINGS
+        .select { |string, _| xml.nodes.first&.value&.include?(string) }
```

Comment:
This will only work if the license comment is the first element - think it should be more flexible?

Reply:
We could load all comments here: `xml.locate('*/^Comment')` I think. Or maybe just top level comments and iterate through each.